ccdaservice/utils/demographics/populate-demographics.js

```diff
@@ -0,0 +1,169 @@
+                },
+            ],
+        },
+    ];
+}
+
+function setNullFlavorIfUnspecifiedOrEmpty(patient, property) {
+    if (patient[property] === DECLINED_TO_SPECIFY || patient[property] === '') {
+        patient[property] = NULL_FLAVOR;
+    }
+}
+
+function getLanguageCode(patient) {
+    return patient.language === 'English'
+        ? 'en-US'
+        : patient.language === 'Spanish'
+            ? 'sp-US'
+            : 'en-US';
+}
+
+function populateDemographics(patient, guardian, documentData) {
+    const oidFacility =
+        documentData.encounter_provider.facility_oid ||
+        '2.16.840.1.113883.19.5.99999.1';
+    const npiFacility =
+        documentData.encounter_provider.facility_npi || NOT_INFORMED;
+
+    setNullFlavorIfUnspecifiedOrEmpty(patient, 'race');
+    setNullFlavorIfUnspecifiedOrEmpty(patient, 'race_group');
+    setNullFlavorIfUnspecifiedOrEmpty(patient, 'ethnicity');
+
+    return {
+        name: {
+            prefix: patient.prefix,
+            suffix: patient.suffix,
+            middle: [patient.mname],
+            last: patient.lname,
+            first: patient.fname,
+        },
+        birth_name: {
+            middle: patient.birth_mname || '',
+            last: patient.birth_lname || '',
+            first: patient.birth_fname || '',
+        },
+        dob: {
+            point: {
+                date: fDate(patient.dob),
+                precision: 'day',
+            },
+        },
+        gender: patient.gender.toUpperCase() || NULL_FLAVOR,
+        identifiers: [
+            {
+                identifier: oidFacility || npiFacility,
+                extension: patient.uuid,
+            },
+        ],
+        marital_status: patient.status.toUpperCase(),
+        addresses: fetchPreviousAddresses(patient),
+        phone: [
+            {
+                number: patient.phone_home,
+                type: 'primary home',
+            },
+            {
+                number: patient.phone_mobile,
+                type: 'primary mobile',
+            },
+            {
+                number: patient.phone_work,
+                type: 'work place',
+            },
+            {
+                number: patient.phone_emergency,
+                type: 'emergency contact',
+            },
+            {
+                email: patient.email,
+                type: 'contact_email',
+            },
+        ],
+        ethnicity: patient.ethnicity || '',
+        race: patient.race || NULL_FLAVOR,
+        race_additional: patient.race_group || NULL_FLAVOR,
+        languages: [
+            {
+                language: getLanguageCode(patient),
+                preferred: true,
+                mode: 'Expressed spoken',
+                proficiency: 'Good',
+            },
+        ],
+        attributed_provider: {
+            identity: [
+                {
+                    root: '2.16.840.1.113883.4.6',
+                    extension: npiFacility,
```

Comment:
You've dropped the ```|| ""``` clause here.  @sjpadgett can this value be "NI" instead of an empty string?

Reply:
Yes, the clause should not be necessary after initializing the value with the proper fallback

Reply:
Looks like you resolved this.